examples/simple/e2e.yaml

```diff
@@ -21,18 +21,38 @@ setup:
   manifests:
     - path: busybox1.yaml
       wait: 
-        - namespace: default
+        - type: kube # wait type may not only kube. it can be `log` or `http` or else.
```

Comment:
I don't think we want to support `log` / `http` in KinD scenarios, it's a best practice to use health check in the pods and eliminating the `type` config

Reply:
Yes, but then in other scenarios, such as docker-compose and run, one might use other health check mechanism. I think add `type` field  in `Wait` struct can let us reuse it in other scenarios? What do you think of it?

Reply:
> Yes, but then in other scenarios, such as docker-compose and run, one might use other health check mechanism. I think add `type` field  in `Wait` struct can let us reuse it in other scenarios? What do you think of it?

docker-compose has its own health check mechanism too, including `log`, `cmd`, etc, so I'd prefer that we put the health check with their corresponding native mechanisms so that we don't bother to create too many configurations in NGE2E (we only check the healthiness exposed by Kubernetes / Docker compose instead of performing the health checks ourselves), also, some of the health checks may need to be run inside the docker containers, if we create `type` like you do in this PR, we may create troubles for ourselves when this happens

Reply:
I see. I will modify this part.